pkg/collectors/infrastructure.go

```diff
@@ -0,0 +1,104 @@
+package collectors
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/kube-state-metrics/pkg/metric"
+	"k8s.io/kube-state-metrics/pkg/version"
+
+	"github.com/golang/glog"
+
+	v1 "github.com/openshift/api/config/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+)
+
+var (
+	descConfigLabelsDefaultLabels = []string{"namespace", "config"}
+
+	infrastructureMetricFamilies = []metric.FamilyGenerator{
+		{
+			Name: "openshift_cluster_infrastructure_provider",
+			Type: metric.MetricTypeGauge,
+			Help: "Reports whether the cluster is configured with an infrastructure provider. type is unset if no cloud provider is recognized or set to the constant used by the Infrastructure config. region is set when the cluster clearly identifies a region within the provider. The value is 1 if a cloud provider is set or 0 if it is unset.",
```

Comment:
Capitalize beginning of sentences.

Reply:
`type` and `region` are the label names, hence they are not capitalized. Added single quotes to highlight them instead